src/dagmc/DagMC.cpp

```diff
@@ -0,0 +1,774 @@
+#include "DagMC.hpp"
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <limits>
+#include <algorithm>
+#include <set>
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <math.h>
+#ifndef M_PI  /* windows */
+# define M_PI 3.14159265358979323846
+#endif
+
+#define MB_OBB_TREE_TAG_NAME "OBB_TREE"
+#define FACETING_TOL_TAG_NAME "FACETING_TOL"
+
+namespace moab {
+
+/* Tolerance Summary
+
+   Facet Tolerance:
+   Maximum distance between continuous solid model surface and faceted surface.
+     Performance:  increasing tolerance increased performance (fewer triangles)
+     Robustness:   should not be affected
+     Knowledge:    user must understand how coarser faceting influences accuracy
+                   of results
+*/
+
+const bool counting = false; /* controls counts of ray casts and pt_in_vols */
+
+// Empty synonym map for DagMC::parse_metadata()
+const std::map<std::string, std::string> DagMC::no_synonyms;
+
+// DagMC Constructor
+DagMC::DagMC(Interface* mb_impl, double overlap_tolerance, double p_numerical_precision) {
+  moab_instance_created = false;
+  // if we arent handed a moab instance create one
+  if (NULL == mb_impl) {
+    mb_impl = new moab::Core();
+    moab_instance_created = true;
+  }
+
+  // set the internal moab pointer
+  MBI = mb_impl;
+
+  // make new GeomTopoTool and GeomQueryTool
+  GTT = new moab::GeomTopoTool(MBI, false);
+  GQT = new moab::GeomQueryTool(GTT, overlap_tolerance, p_numerical_precision);
+
+  // This is the correct place to uniquely define default values for the dagmc settings
+  defaultFacetingTolerance = .001;
+}
+
+// Destructor
+DagMC::~DagMC() {
+  // delete the GeomTopoTool and GeomQueryTool
+  delete GTT;
```

Comment:
Rather than manually `new`'ing and `deleting` these objects, I would recommend using a RAII ("Resource Allocation Is Initialization") technique to manage their memory [1].  Specifically, you can use smart pointers [2] such as `std::unique_ptr` or `std::shared_ptr` to allocate `GTT` and `GQT` in the constructor, and subsequently not be responsible for their deletion.

Unless you are allocating a computationally large number of pointers (such as in a core data structure in a game engine), it's typically recommended to use a smart pointer when possible to guard against memory leaks.

[1] https://en.cppreference.com/w/cpp/language/raii
[2] https://en.cppreference.com/book/intro/smart_pointers